src/silx/gui/hdf5/Hdf5TreeModel.py

```diff
@@ -557,7 +564,7 @@ def synchronizeIndex(self, index):
     def h5pyObjectRow(self, h5pyObject):
         for row in range(self.__root.childCount()):
             item = self.__root.child(row)
-            if item.obj == h5pyObject:
+            if hash(item.obj) == hash(h5pyObject):
```

Comment:
Same as the other PR. I don't think it is needed.
Or you are about to change the previous behaviour, which is maybe not bad, i can't tell.

Reply:
It was removed, instead `__eq__` was rewritten to NOT compare the data but the filename/dataset name instead.